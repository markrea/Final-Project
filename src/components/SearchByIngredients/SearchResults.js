import React, { useState, useEffect } from "react";
import axios from "axios";
import PropTypes from "prop-types";
import "../../styles/SearchResults.css";

const API_KEY2 = "48aba1bc5bce4ba3901fc4b1e3591949";
const Result = ({ results }) => {
  const [href, setHref] = useState("");
  useEffect(() => {
    axios
      .get(
        `https://api.spoonacular.com/recipes/${results.id}/information?apiKey=${API_KEY2}`
      )
      .then((response) => {
        const link = response.data.sourceUrl;
        return setHref(link);
      })
      .catch((err) => console.log(err));
  }, []);
  return (
    <div key={results.image} className="innerSearchResult">
      <img src={results.image} alt="mockAlt" className="card-photo" />
      <figcaption className="caption">{results.title}</figcaption>
      <a href={href} target="_blank" rel="noopener noreferrer" >
        More Details
      </a>
    </div>
  );
};
const SearchResults = ({ searchResults }) => {
  return (
    <div className="SearchResults" data-testid="results-div">
      <div className="test">
        {searchResults.map((results) => (
          <Result key={results.id} results={results} />
        ))}
      </div>
    </div>
  );
};
SearchResults.propTypes = {
  Result: PropTypes.array,
};
export default SearchResults;
