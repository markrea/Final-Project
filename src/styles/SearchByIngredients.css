
.FormText{
   
    background: rgba(214, 203, 203, 0.507);
    margin: 0 auto;
    padding: 0;
    width: 95%;
    border-radius: 10px; 

}

 h6{
    color: rgb(68, 67, 67);
    font-family: 'Josefin Slab', serif;
    font-size: 2rem;
}