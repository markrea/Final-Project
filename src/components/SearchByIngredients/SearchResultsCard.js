import React from "react";
import SearchResults from "./SearchResults";
import "../../styles/App.css";

const SearchResultsCard = ({searchResults}) => {
  return (
    <div>
      <div className="search-results" data-testid="search-div">
        {searchResults ? (
          <SearchResults searchResults={searchResults} />
        ) : (
          <div>
            <h2 className="results-text"></h2>
          </div>
        )}
      </div>
    </div>
  );
};
export default SearchResultsCard;
