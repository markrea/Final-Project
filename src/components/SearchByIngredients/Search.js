import React, { useState } from "react";
import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

import getRecipes from "../../requests/getRecipes";

import "../../styles/Search.css";
import SearchForm from "./SearchForm";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState("");
  const [diet, setDiet] = useState("");
  const [intolerances, setIntolerances] = useState([]);
  const [loading, setLoading] = useState(false);

  const handleSubmit = async (event) => {
    setLoading(true);
    event.preventDefault(); // stops browser reloading
    setSearchResults(await getRecipes(value, diet, intolerances));
    setLoading(false);
  };

  const handleChange = async (e) => {
    setValue(e.target.value);
  };

  const handleSelect = async (e) => {
    setDiet(e.target.value);
  };

  return (
    <div className="searchFormDiv" data-testid="searchFormDiv">
      <SearchForm
        handleChange={handleChange}
        handleSubmit={handleSubmit}
        handleSelect={handleSelect}
        query={value}
        diet={diet}
        setIntolerances={setIntolerances}
      />
      {loading && (
        <Loader
          className="loader"
          type="ThreeDots"
          color="green"
          height={60}
          width={60}
          timeout={3000}
        />
      )}
    </div>
  );
};
export default Search;
//  <i className="fa fa-search" aria-hidden="true" />
