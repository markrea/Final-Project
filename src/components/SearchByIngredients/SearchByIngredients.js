import React, { useState } from "react";
import Search from "./Search";
import SearchResultsCard from "./SearchResultsCard";

import "../../styles/SearchByIngredients.css";
import "../../styles/HomePage.css";

const SearchByIngredients = () => {
  const [searchResults, setSearchResults] = useState(null);

  return (
    <div className="SearchByIngredients">
      <div className="SearchByIngredientsText">
        <h2>Search By Ingredients</h2>
      </div>
      <div className="FormText">
        <div className="mainText">
          <h4>Enter your ingredients below to see what you can make.</h4>
        </div>

        <Search setSearchResults={setSearchResults} />
        <br />
        <br />
        <br />
        <SearchResultsCard searchResults={searchResults} />
      </div>
    </div>
  );
};
export default SearchByIngredients;
