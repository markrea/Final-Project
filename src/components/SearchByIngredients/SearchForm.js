import React from "react";
import "../../styles/Search.css";
import Select from "react-select";
import Proptypes from "prop-types";

const SearchForm = ({
  handleSubmit,
  handleChange,
  handleSelect,
  query,
  diet,
  setIntolerances,
}) => {
  const options = [
    { value: "dairy", label: "Dairy" },
    { value: "egg", label: "Egg" },
    { value: "gluten", label: "Gluten" },
    { value: "peanut", label: "Peanut" },
    { value: "seafood", label: "Seafood" },
    { value: "sesame", label: "Sesame" },
    { value: "shellfish", label: "Shellfish" },
    { value: "soy", label: "Soy" },
    { value: "treenut", label: "Tree Nut" },
  ];

  return (
    <form
      type="text"
      className="input-form"
      onSubmit={handleSubmit}
      data-testid="searchForm"
    >
      <fieldset>
        <select
          id="diet"
          name="diet"
          size="1"
          value={diet}
          placeholder="Select Diet"
          onChange={handleSelect}
        >
          <option defaultValue="null">Select Diet</option>
          <option value="vegetarian">Vegetarian</option>
          <option value="vegan">Vegan</option>
          <option value="glutenfree">Gluten Free</option>
          <option value="pescatarian">Pescastarian</option>
        </select>
        <Select
          isMulti
          autoFocus
          name="colors"
          options={options}
          className="select-container"
          classNamePrefix="select"
          placeholder="Select intolerances"
          onChange={setIntolerances}
          theme={(theme) => ({
            ...theme,
            borderRadius: "10px",
            backgroundColor: "rgba(196, 208, 162, 1)",
          })}
        />
        <br />

        <input
          id="search"
          name="recipeName"
          type="text"
          placeholder="Enter ingredients"
          onChange={handleChange}
          value={query}
          data-testid="input-id"
        />
        <button type="submit" className="submit">
          Go!
        </button>
        <br />
      </fieldset>
    </form>
  );
};

SearchForm.propTypes = {
  handleSubmit: Proptypes.func.isRequired,
  handleChange: Proptypes.func.isRequired,
  handleSelect: Proptypes.func.isRequired,
  query: Proptypes.string.isRequired,
  diet: Proptypes.string.isRequired,
  setIntolerances: Proptypes.string.isRequired,
};

export default SearchForm;
