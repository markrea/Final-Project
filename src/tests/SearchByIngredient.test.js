import React from "react";
import { render } from "@testing-library/react";
import SearchByIngredients from "../components/SearchByIngredients/SearchByIngredients";
import "@testing-library/jest-dom/extend-expect";

describe("SearchByIngredients", () => {
  it("renders component correctly", () => {
    const { asFragment } = render(<SearchByIngredients />);
    expect(asFragment).toMatchSnapshot();
  });
  it("displays a search by ingredients div", () => {
    const { getByTestId } = render(<SearchByIngredients />);
    expect(getByTestId("searchByIngredientsDiv")).toBeInTheDocument();
  });
});
