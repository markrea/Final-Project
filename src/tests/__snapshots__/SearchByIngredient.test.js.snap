// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByIngredients renders component correctly 1`] = `[Function]`;
